### DISCUSSION TITLE
Logo and Favicon as part of the codebase

### DISCUSSION THREAD

**Discussion by juancarlosjr97** [directus/directus#8906] [Archive]
Related discussions: 
* https://github.com/directus/directus/discussions/8354
* https://github.com/directus/directus/discussions/4184
* https://github.com/directus/v8-archive/discussions/2359

In the current project, I am working on, the database is version controlled using flyway. However, one the things I cannot version control is the logo (also the favicon but is not that important) to create clean installations with everything needed.

I am thinking if a conditional in the code with a variable pointing such as `LOGO_NAME` as `logo.png` and if the logo field in the row of `directus_settings` is empty, it will serve that logo.

**Comment by rijkvanzanten** [comment]
I'd rather not have two different configuration locations for a single setting (like project logo in both directus_settings _and_ the environment variables), as that might become confusing pretty fast. Pulling the file path from an env var might break as well, as that file might not be publicly accessible 🤔

**Reply by juancarlosjr97** [reply]
What about if we have a logo image that can be hosted locally in the uploads directory that can be versioned, but everything else is server either by external storage or even from the host but the logo I think moving forward has to be versioned.

I have not tried it, but what about if I save an image locally, then I changed to S3 storage (which is what I am using) to keep the logo versioned?

**Reply by rijkvanzanten** [reply]
> What about if we have a logo image that can be hosted locally in the uploads directory that can be versioned, but everything else is server either by external storage or even from the host but the logo I think moving forward has to be versioned.

You might be able to do that already actually! The API allows for multiple storage adapters simultaneously, so you could configure S3 as the first one (making it the one the app uses), and attach a secondary local one for the files you want versioned 🤔

**Reply by juancarlosjr97** [reply]
That sounds great @rijkvanzanten!

How do we attach secondary storage??? I  am happy to test it and get back here later :)

**Reply by rijkvanzanten** [reply]
The STORAGE_LOCATIONS env var accepts a CSV of values, so you should be able to do:

```
STORAGE_LOCATIONS="s3,local"

STORAGE_S3_A="B"
STORAGE_LOCAL_X="Y"
```

(where a/x are the actual config flags you want obv)

**Reply by juancarlosjr97** [reply]
I have to say, that works perfectly and I love it :) 

Nicely done guys!
